[EE11] [WFLY-19590] [MP Platform 7] MicroProfile Telemetry 2.0

jasondlee opened the pull request:
Update otel artifact names and versions
Move changes to WF Preview

____

<--- THIS SECTION IS AUTOMATICALLY GENERATED BY WILDFLY GITHUB BOT. ANY MANUAL CHANGES WILL BE LOST. --->

> Wildfly issue links:
> * [WFLY-19590](https://issues.redhat.com/browse/WFLY-19590)

<--- END OF WILDFLY GITHUB BOT REPORT --->

More information about the [wildfly-bot[bot]](https://github.com/wildfly/wildfly-github-bot)

bstansberry commented on preview/galleon-local/src/main/resources/modules/system/layers/base/io/opentelemetry/api/main/module.xml:
```diff
@@ -0,0 +1,32 @@
+        <module name="io.smallrye.opentelemetry" services="export"/>
+
+        <module name="java.logging"/>
+        <module name="jdk.management"/>
```
@jasondlee It looks like java.management is needed as well.

https://ci.wildfly.org/viewLog.html?buildId=460715&buildTypeId=WF_PullRequestEe11_BootableJarLinuxJdk17&fromSakuraUI=true

```
Caused by: java.lang.NoClassDefFoundError: java/lang/management/ManagementFactory
	at io.opentelemetry.api//io.opentelemetry.instrumentation.runtimemetrics.java8.Classes.registerObservers(Classes.java:43)
	at io.smallrye.opentelemetry@2.8.0//io.smallrye.opentelemetry.implementation.cdi.OpenTelemetryProducer.getOpenTelemetry(OpenTelemetryProducer.java:51)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.jboss.weld.core@6.0.0.Beta4//org.jboss.weld.injection.StaticMethodInjectionPoint.invoke(StaticMethodInjectionPoint.java:100)
	... 82 more
Caused by: java.lang.ClassNotFoundException: java.lang.management.ManagementFactory from [Module "io.opentelemetry.api" from local module loader @be64738 (finder: local module finder @3ba9ad43 (roots: /opt/buildAgent/work/a2ee6b14096e9408/testsuite/integration/microprofile/target/wildfly/modules,/opt/buildAgent/work/a2ee6b14096e9408/testsuite/integration/microprofile/target/wildfly/modules/system/layers/base))]
	at org.jboss.modules.ModuleClassLoader.findClass(ModuleClassLoader.java:200)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClassUnchecked(ConcurrentClassLoader.java:408)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClass(ConcurrentClassLoader.java:396)
	at org.jboss.modules.ConcurrentClassLoader.loadClass(ConcurrentClassLoader.java:115)
	... 89 more
```

jasondlee replied:
Updated.

jasondlee committed:
[EE11] [WFLY-19590] [MP Platform 7] MicroProfile Telemetry 2.0

Update otel artifact names and versions for Preview
Update BOMs to reflect split between Standard and Preview
Duplicate models under Preview to update for 2.0
Add (temporary) check for Preview to set up exports as appropriate
Modify preview test profiles to test 2.0 instead of 1.1
